minStack.py

```diff
@@ -0,0 +1,39 @@
+class MinStack(object):
+
+    def __init__(self):
+        """
+        initialize your data structure here.
+        """
+        self.stack = []
+        self.minstack = []
+
+    def push(self, x):
+        """
+        :type x: int
+        :rtype: void
+        """
+        self.stack.append(x)
+        if len(self.minstack) == 0 or x <= self.minstack[-1]:
+            self.minstack.append(x)
+        
+
+    def pop(self):
+        """
+        :rtype: void
+        """
+        if self.stack[-1] == self.minstack[-1]:
```

Comment:
Shouldn't the last item of `minStack` be popped out always?
Why there is a condition here

Reply:
last item of minStack is popped out only if it equals to last item of stack